docs/docker_host_ssh.md

```diff
@@ -0,0 +1,116 @@
+Reuse docker host SSH daemon
+============================
+
+It is possible to use the SSH daemon that runs on the docker host instead of using a separate SSH port for GitLab.
+
+## Setup
+
+Create the system `git` user if necessary (make sure it has a working shell):
+
+```bash
+useradd --system git --home-dir /srv/docker/gitlab/data
+```
+
+The home directory must point to GitLab data directory, as it takes `.ssh/authorized_keys` from there.
+
+Now create `gitlab-shell` home directory at the same location as in the container:
+
+```bash
+mkdir -p /home/git/gitlab-shell/bin
```

Comment:
The guide is written with this alternate gitlab docker container in mind. That said, the primary different to the omnibus docker container is probably just different paths. The path on the host must match the path to gitlab-shell inside the container, thus using `/var/opt/gitlab/gitlab-shell/bin` should work.
I'm, however, not sure which uid/gid for the git user is used by the omnibus docker container. As the host machine should ideally use the same uid/gid for its git user, this may cause further problems.